### ISSUE TITLE
deprecate twisted.enterprise.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3022]
|[<img alt="bitglue's avatar" src="https://avatars.githubusercontent.com/u/1232472?s=50" width="50" height="50">](https://github.com/bitglue)| @bitglue reported|
|-|-|
|Trac ID|trac#3022|
|Type|enhancement|
|Created|2008-02-06 19:24:07Z|
|Branch|https://github.com/twisted/twisted/tree/deprecated-enterprise-3022|

safe() doesn't actually always make things safe. quote() is horribly misguided in it's attempt to add quotes to some things and not others. All the other components of this module seem to be parts of other deprecated modules. Also, DBAs might burst in flames if they look at the module closely, which is a huge liability.

<details><summary>Searchable metadata</summary>

```
trac-id__3022 3022
type__enhancement enhancement
reporter__indigo indigo
priority__highest highest
milestone__ 
branch__branches_deprecated_enterprise_3022 branches/deprecated-enterprise-3022
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1202325847000000 1202325847000000
changetime__1203612778000000 1203612778000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph set owner to @bitglue|
|-|-|

Pretty much everything in twisted.enterprise (except adbapi) should have large, aggressive warnings in module docstrings explaining they are deprecated, and not deprecated in favor of anything, just nobody should use them ever.  Every function in util should, similarly, emit a deprecation warning if you were to call it.

Would you care to contribute a patch which does this?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [22503]) Branching to 'deprecated-enterprise-3022'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

 * test_entreprise prints a warning about util when run, this warning should be removed.
 * test_safe and test_getKeyColumn should probably named test_safeDeprecation and test_getKeyColumnDeprecation
 * 2.6.0 could be replaced with 8.0.0, unless I miss something.
 * the warning in util show that the deprecation API is not yet complete. Maybe we could open a ticket for a 'deprecate module' facility.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun removed owner|
|-|-|

I renamed the test methods.  For the rest, here are my thoughts:

  * `twisted.enterprise.util`, the module, is deprecated, so there should be a deprecation emitted when it is imported.  In some future release, we'll delete the whole file, so code which imports it (even if it doesn't use anything from it) will break, so there should be a warning.  I dunno what to do about the warning being emitted by the test suite.  It's a problem that shows up in a number of places in Twisted.
  * 2.6.0 could be replaced with 8.0.0, but plenty of other places in Twisted say 2.6.0 now.  Since the release person will have to update some of these numbers anyway, I don't think it's a big deal if new changes use 2.6.0 as well.
  * Sounds like a good thing, though I can't think of any way it could work at the moment.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_8"></a>|@dreid set owner to @exarkun|
|-|-|

I agree about the warning in util.py, but I also don't know how it should work so I don't think it should hold up this ticket.

I do feel like the _unreleasedVersion and _unreleasedDeprecation are a little wonky, but I can't think of a better way to do it.  (A module that just holds deprecation functions for twisted versions perhaps?... I don't know.)

Go ahead and merge both of these issues are for the deprecation framework and shouldn't delay deprecation of something as horrendous as t.e.util.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [22677]) Merge deprecated-enterprise-3022

Author: exarkun
Reviewer: therve, dreid
Fixes [#3022](https://github.com/twisted/twisted/issues/3022)
Refs [#3040](https://github.com/twisted/twisted/issues/3040)

Deprecate twisted.enterprise.util and everything in it.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
